lib/sink.js

```diff
@@ -203,11 +223,22 @@ module.exports = class SinkGCS extends EventEmitter {
         );
 
         await this[ready];
-        await this[getGcsFile](fileName).save(fileContent, {
-            metadata: {
-                contentType,
-            },
-        });
+        const file = this[getGcsFile](fileName);
+        const saveFile = () =>
+            file.save(fileContent, { metadata: { contentType } });
+        try {
+            return await pRetry(saveFile, { retries: 3 });
```

Comment:
I'm not sure we should retry save, in the same way you should not retry `POST`. @ivarconr thoughts?

Reply:
k. We did see cases of upload failures though so we would back at not fully handling our issues.

Reply:
we have retry in the client on the other side, though.

having multiple layers of retry doesn't scale.

Reply:
yeah. true dat

Reply:
How should we resolve this one? Ask @ivarconr for comment again?